### ISSUE TITLE
How to train with constant tensor? (RNN classification with attention and masking)

### ISSUE THREAD

**Issue by tmchojo** [keras-team/keras#9107]
self.inputs = Input(shape=(self.N_TIMESTEP,))    ## batch, time(int)
    E = Embedding(self.N_FEATURE[0], self.N_EMB, mask_zero=True, input_length=self.N_TIMESTEP)(self.inputs)   ## batch, time, emb
         
    # LSTM
    D0 = Dropout(0.2)(E)
    L1 = LSTM(self.N_HIDDEN, activation='tanh', return_sequences=True)(D0) ## batch, time, hidden
    D1 = Dropout(0.2)(L1)
    L2 = LSTM(self.N_HIDDEN, activation='tanh', return_sequences=True)(D1)
    D2 = Dropout(0.2)(L2)
    L3 = LSTM(self.N_HIDDEN, activation='tanh', return_sequences=True)(D2)
     
    # attention
    att_vec0 = TimeDistributed(Dense(1))(L3) ## batch, time, 1
    att_vec1 = Lambda(lambda x: K.mean(x, axis=2))(att_vec0) ## batch, time
        # remove padding from attention
    att_vec2 = Lambda(lambda x: K.clip(x, 0, 1))(self.inputs) # 1 1 0 0 0
    att_vec3 = Add()([self.att_vec2, K.constant(-1, shape=(self.N_BATCH, self.N_TIMESTEP))]) # 0 0 -1 -1 -1
    att_vec4 = Multiply()([att_vec3, K.constant(2**15, shape=(self.N_BATCH, self.N_TIMESTEP))]) # 0 0 -2^15 -2^15 -2^15
    att_vec7 = Add()([att_vec1, att_vec4]) # b1 b2 -2^15 -2^15 -2^15
        #
    self.att_vec = Activation('softmax')(att_vec7) # a1 a2 0 0 0
    att_vec8 = RepeatVector(self.N_HIDDEN)(self.att_vec) ## batch, hidden, time
    att_vec9 = Permute((2, 1))(att_vec8) ## batch, time, hidden
    
    # prediction
    C0 = Multiply()([L3, att_vec9]) ## batch, time, hidden
    C = Lambda(lambda x: K.sum(x, axis=1))(C0) ## batch, hidden
    outputs = Dense(self.N_CLASS, activation='softmax')(C) ## batch, class

I want to remove padding timesteps from attention.
So, I make values of padding parts -2^15 before softmax.
For this, I use **K.constant**.
But Keras doesn't work with this **K.constant**. ('Tensor' object has no attribute '_keras_history' error)
Is there another way to define constant tensor in Keras?
